librz/core/cmd/cmd_print.c

```diff
@@ -2330,105 +2313,22 @@ static bool cmd_print_pxA(RzCore *core, int len, RzOutputMode mode) {
 	return true;
 }
 
-static ut8 *old_transform_op(RzCore *core, const char *val, char op, int *buflen) {
-	RzCoreWriteOp wop;
-	switch (op) {
-	case '2':
-		wop = RZ_CORE_WRITE_OP_BYTESWAP2;
-		break;
-	case '4':
-		wop = RZ_CORE_WRITE_OP_BYTESWAP4;
-		break;
-	case '8':
-		wop = RZ_CORE_WRITE_OP_BYTESWAP8;
-		break;
-	case 'a':
-		wop = RZ_CORE_WRITE_OP_ADD;
-		break;
-	case 'A':
-		wop = RZ_CORE_WRITE_OP_AND;
-		break;
-	case 'd':
-		wop = RZ_CORE_WRITE_OP_DIV;
-		break;
-	case 'l':
-		wop = RZ_CORE_WRITE_OP_SHIFT_LEFT;
-		break;
-	case 'm':
-		wop = RZ_CORE_WRITE_OP_MUL;
-		break;
-	case 'o':
-		wop = RZ_CORE_WRITE_OP_OR;
-		break;
-	case 'r':
-		wop = RZ_CORE_WRITE_OP_SHIFT_RIGHT;
-		break;
-	case 's':
-		wop = RZ_CORE_WRITE_OP_SUB;
-		break;
-	case 'x':
-		wop = RZ_CORE_WRITE_OP_XOR;
-		break;
-	default:
-		wop = RZ_CORE_WRITE_OP_XOR;
-		rz_warn_if_reached();
-		break;
-	}
-
+/* Uses data from clipboard if value is NULL */
+static bool print_operation_transform(RzCore *core, RzCoreWriteOp op, RZ_NULLABLE const char *val) {
 	ut8 *hex = NULL;
-	int hexlen = -1;
+	int hexlen = -1, buflen = -1;
 	if (val) {
-		val = rz_str_trim_head_ro(val);
 		hex = RZ_NEWS(ut8, (strlen(val) + 1) / 2);
 		if (!hex) {
-			return NULL;
+			return false;
 		}
-
 		hexlen = rz_hex_str2bin(val, hex);
 	}
-	ut8 *result = rz_core_transform_op(core, core->offset, wop, hex, hexlen, buflen);
+	ut8 *buf = rz_core_transform_op(core, core->offset, op, hex, hexlen, &buflen);
```

Comment:
can you do a small refactor and change the `int` for all the buffer sizes to `size_t` (including in the function declaration)?

Reply:
Done, see second commit.